falcon_multiqc/commands/sql.py

```diff
@@ -0,0 +1,52 @@
+import click
+import glob
+import os
+from sqlalchemy import create_engine
+from database.config import DATABASE_URI
+from database.process_query import run_multiqc
+
+# example query: 
+
+# this query will query fastqc tool for Total seqence metric - since only the first batch ran fastqc it will only return first batch
+# SELECT s.sample_name, b.path FROM public.raw_data as rd join sample as s on (s.id = rd.sample_id) join batch as b on (s.batch_id = b.id) where rd.qc_tool = 'fastqc' AND (rd.metrics->>'Total Sequences')::numeric::integer >= 316624794;
+
+# this queries two different tools (picard_insertsize and verifybamid) and on of their respective metrics - returns sampleID from first batch and sampleID from 4012 batch 
+# SELECT s.sample_name, b.path FROM public.raw_data as rd join sample as s on (s.id = rd.sample_id) join batch as b on (s.batch_id = b.id) where ((rd.qc_tool = 'picard_insertSize' AND (rd.metrics->>'MEAN_INSERT_SIZE')::numeric::integer <= 390) OR (rd.qc_tool = 'verifybamid' AND (rd.metrics->>'#READS')::numeric::integer >= 4850000));
+
+@click.command()
+@click.option("-d", "--directory", type=click.STRING, required=True, help="where query result will be saved")
+@click.option("-s", "--sql", type=click.INT, required=False, help="Enter 1 to start SQL querying tool") 
+def cli(directory, sql):
+    if sql == 1:
+        """SQL query tool: ensure all queries SELECT for sample_name from sample table AND path from batch table"""
+
+        count = 0 # to ensure new file name generated
+        config = os.path.abspath("./database/falconqc.config") # loading falconqc config file 
+        query_eng = create_engine(DATABASE_URI) # establishing engine for communication 
```

Comment:
Can we remove this, and instead make a method in crud.py that will return a query engine? I.e. get_query_engine() which simply returns engine.connect() or similar. crud.py already creates an engine.

Reply:
I've changed it up so I'm just important the literal engine created in crud, it works fine.. I dispose of it at the end which still works fine, I'm not sure what's optimal lol